Hestia/Hestia/DevicesScreen/AddDeviceScreen/TableSourceAddDeviceDevice.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using UIKit;
+using Foundation;
+
+using Hestia.DevicesScreen.resources;
+using System.Drawing;
+using System.Collections;
+using Hestia.backend;
+using Hestia.backend.models;
+
+namespace Hestia.DevicesScreen
+{
+    public class TableSourceAddDeviceDevice: UITableViewSource
+    {
+        // The viewController to which the TableView connected to this Source lives in
+        UITableViewControllerAddDeviceDevice owner;
+
+        // List with plugins and collection they belong to
+        List<string> plugins;
+        string collection;
+
+        // The kind of cell that is used in the table (set in Storyboard)
+        string CellIdentifier = "tableCellDeviceType";
+
+        // Constructor. Gets the device data and the ViewController
+        public TableSourceAddDeviceDevice(List<string> plugins,
+                                          string collection,
+                    UITableViewControllerAddDeviceDevice owner)
+        {
+            this.plugins = plugins;
+            this.owner = owner;
+            this.collection = collection;
+
+        }
+
+        // We have only one section with devices (thus far)
+        public override nint NumberOfSections(UITableView tableView)
+        {
+            return 1;
```

Comment:
If the number of sections is hardcoded, do it in some config file or constant, don't let a function always return 1.